community/kernel/src/main/java/org/neo4j/kernel/impl/store/counts/CountsTracker.java

```diff
@@ -300,56 +449,163 @@ public void close()
         }
     }
 
-    public void rotate( long lastCommittedTxId ) throws IOException
+    @Override
+    protected Object parseHeader( String key, ReadableBuffer value )
```

Comment:
We could replace the `String` used for the key definitions (in the constructor of `AbstractKeyValueStore`) with something that is type safe. That could actually be quite nice, I'll give that a try.